CorsixTH/Lua/dialogs/resizables/menu_list_dialogs/custom_campaign.lua

```diff
@@ -80,6 +101,13 @@ function UICustomCampaign:UICustomCampaign(ui)
 
   self.description_offset = 0
 
+  -- Warn about hidden duplicate campaigns
+  if duplicates then
```

Comment:
You recognize duplicates while reading, and then report it here.

If you read and return every valid campaign you find, and here check the results for duplicates, you separate duplicate handling from reading campaigns.

Reply:
Eg extending to "3 similarly named campaigns X were hidden" becomes simpler then, as doesn't need to be done inter-mixed with reading and validating.

That is, doing one thing at a time often makes extending functionality tomorrow simpler.